File: client/src/services/jobService.js
import axiosClient from "../api/axiosClient";

export const getJobs = async (filters = {}) => {
  const response = await axiosClient.get("/jobs", {
    params: {
      keyword: filters.keyword || undefined,
      location: filters.location || undefined,
      employmentType: filters.employmentType || undefined,
      experienceLevel: filters.experienceLevel || undefined,
    },
  });

  return response.data;
};

export const getJobById = async (jobId) => {
  const response = await axiosClient.get(`/jobs/${jobId}`);
  return response.data;
};

File: backend/src/main/java/com/careeros/service/JobService.java
package com.careeros.service;

import com.careeros.dto.response.JobDetailsResponse;
import com.careeros.dto.response.JobResponse;

import java.util.List;

public interface JobService {

    List<JobResponse> filterJobs(
            String keyword,
            String location,
            String employmentType,
            String experienceLevel
    );

    JobDetailsResponse getJobById(String jobId);

}